backend/gentxs.go

```diff
@@ -0,0 +1,654 @@
+	tx := std.Tx{
+		Msgs: []std.Msg{
+			vm.MsgRun{
+				Caller: creator,
+				Send:   []std.Coin{},
+				Package: &gnovm.MemPackage{
+					Name:  "main",
+					Files: []*gnovm.MemFile{{Name: "main.gno", Body: body}},
+				},
+			},
+		},
+		Fee: std.Fee{
+			GasWanted: 10000000,
+			GasFee:    std.NewCoin("ugnot", 1000000),
+		},
+	}
+
+	return gnoland.TxWithMetadata{
+		Tx: tx,
+		Metadata: &gnoland.GnoTxMetadata{
+			Timestamp: poll.CreatedAt.Unix(),
+		},
+	}, nil
+}
+
+func createEventRegTx(chain *gnoZenaoChain, event *zeni.Event, caller cryptoGno.Address) (gnoland.TxWithMetadata, error) {
+	eventPkgPath := chain.eventRealmPkgPath(event.ID)
+	tx := std.Tx{
+		Msgs: []std.Msg{
+			vm.MsgCall{
+				Caller:  caller,
+				Send:    []std.Coin{},
+				PkgPath: chain.eventsIndexPkgPath,
+				Func:    "IndexEvent",
+				Args:    []string{eventPkgPath},
+			},
+		},
+		Fee: std.Fee{
+			GasWanted: 10000000,
+			GasFee:    std.NewCoin("ugnot", 1000000),
+		},
+	}
+
+	return gnoland.TxWithMetadata{
+		Tx: tx,
+		Metadata: &gnoland.GnoTxMetadata{
+			Timestamp: event.CreatedAt.Unix() + 1, // +1 to avoid collision with event creation
+		},
+	}, nil
+}
+
+func createEventRealmTx(db zeni.DB, chain *gnoZenaoChain, event *zeni.Event, creator cryptoGno.Address, organizersIDs []string, privacy *zenaov1.EventPrivacy) (gnoland.TxWithMetadata, error) {
+	organizersAddr := mapsl.Map(organizersIDs, chain.UserAddress)
+	eRealm, err := genEventRealmSource(organizersAddr, creator.String(), genTxsConf.name, &zenaov1.CreateEventRequest{
+		Title:       event.Title,
+		Description: event.Description,
+		ImageUri:    event.ImageURI,
+		Location:    event.Location,
+	}, privacy)
+	if err != nil {
+		return gnoland.TxWithMetadata{}, err
+	}
+
+	eventPkgPath := chain.eventRealmPkgPath(event.ID)
+	eventTx := std.Tx{
+		Msgs: []std.Msg{
+			vm.MsgAddPackage{
+				Creator: creator,
+				Deposit: []std.Coin{},
+				Package: &gnovm.MemPackage{
+					Name:  "event",
+					Path:  eventPkgPath,
+					Files: []*gnovm.MemFile{{Name: "event.gno", Body: eRealm}},
+				},
+			},
+		},
+		Fee: std.Fee{
+			GasWanted: 10000000,
+			GasFee:    std.NewCoin("ugnot", 1000000),
+		},
+	}
+
+	return gnoland.TxWithMetadata{
+		Tx: eventTx,
+		Metadata: &gnoland.GnoTxMetadata{
+			Timestamp: event.CreatedAt.Unix(),
+		},
+	}, nil
+}
+
+func createParticipationRegTx(chain *gnoZenaoChain, event *zeni.Event, caller cryptoGno.Address, participantAddr string) (gnoland.TxWithMetadata, error) {
+	eventPkgPath := chain.eventRealmPkgPath(event.ID)
+	tx := std.Tx{
+		Msgs: []std.Msg{
+			vm.MsgCall{
+				Caller:  caller,
+				Send:    []std.Coin{},
+				PkgPath: chain.eventsIndexPkgPath,
+				Func:    "AddParticipant",
+				Args:    []string{eventPkgPath, participantAddr},
+			},
+		},
+		Fee: std.Fee{
+			GasWanted: 10000000,
+			GasFee:    std.NewCoin("ugnot", 1000000),
+		},
+	}
+
+	return gnoland.TxWithMetadata{
+		Tx: tx,
+		Metadata: &gnoland.GnoTxMetadata{
+			Timestamp: event.CreatedAt.Unix(),
```

Comment:
shouldn't this be the timestamp of the ticket creation?
should be at least `event.CreatedAt + 2` since we register the event at `event.CreatedAt + 1`, no?